parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReader.java

```diff
@@ -76,18 +87,12 @@ public void initFromPage(int valuesCount, ByteBufferInputStream stream)
       throw new ParquetDecodingException(errorMessage);
     }
 
-    // Allocate buffer for all of the byte stream data.
+    // Eagerly read and decode the data. This allows returning stable
+    // Binary views into the internal decode buffer for FIXED_LEN_BYTE_ARRAY.
     final int totalSizeInBytes = stream.available();
-    byteStreamData = new byte[totalSizeInBytes];
-
-    // Eagerly read the data for each stream.
-    final int numRead = stream.read(byteStreamData, 0, totalSizeInBytes);
-    if (numRead != totalSizeInBytes) {
-      String errorMessage = String.format(
-          "Failed to read requested number of bytes. Expected: %d. Read %d.", totalSizeInBytes, numRead);
-      throw new ParquetDecodingException(errorMessage);
-    }
-
+    final ByteBuffer encodedData = stream.slice(totalSizeInBytes).slice(); // possibly zero-copy
+    final byte[] decodedData = decodeData(encodedData, this.valuesCount);
```

Comment:
Do we need to hold the reference of `decodedData`?

Reply:
I assume [`ByteBuffer.wrap`](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#wrap-byte:A-) stores a reference to it, otherwise those APIs would be marked as unsafe and [`ByteBuffer.array`](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#array--) wouldn't possibly work.